common/services/user-locations.js

```diff
@@ -0,0 +1,68 @@
+const axios = require('axios')
+
+const { AUTH_PROVIDERS, NOMIS_ELITE2_API } = require('../../config')
+const { decodeAccessToken } = require('../lib/access-token')
+const referenceDataService = require('./reference-data')
+
+async function getUserLocations(token) {
```

Comment:
After looking at the other methods in this module I think this might be neater if its restructured to return a promise as well rather than use async/await.

Reply:
🤔 I've refactored this as you suggest and it's cleaned it up a bit, though it leaves a somewhat nasty nested `then` which I think makes it a bit less clear what the return value of the function is. Have a look and let me know what you think.